### PR TITLE
Refactored sample network test cases for errors in api calls

### PR DESCRIPTION
**Pull request by rewati-arista**
Additionally added missing logging, and applied consistent dut filters for all test cases

# Please include a summary of the changes

* Missing logs
  (1) test_environment.py
  (2) test_tacacs.py
  (3) test_users.py
  
* Incorrect object reference
  (1) test_daemon.py: error being thrown in case of empty object, handled it accordingly
  
* Incorrect pasisng of arguments to api
(1) test_aaa.py
(2) test_dns.py

* Added skip logic
  (1) test_environment.py

* Changed filters for test definitions to be consistent with Vane Demo Lab
(1) test_definition.yaml files for api, cpu, daemon, host, and lldp tests

# Any specific logic/part of code you need extra attention on

test_daemon.py refactoring

# Include the Issue number and link

#720 

# List/Attach any dependencies/past issues that are required for this change/provide context

Further issue created #725  for refactoring commands referenced in ntp, system, users, environment, log, and dns test case
Before closing #720 need to fix vane_system_tests using #727 as well

# Type of change

- - [X] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [X] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [X] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure

### CODE REVIEW

File: sample_network_tests/tacacs/test_tacacs.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def test_if_tacacs_is_sending_messages_on_(self, dut, tests_definitions):
             self.output = (
                 tops.output_msg
             ) = tops.comment = f"\nRouter {tops.dut_name} does not have TACACS servers configured"
+            logging.info(f"\nRouter {tops.dut_name} does not have TACACS servers configured")
```
Comment (kgrozis-arista) [inline sample_network_tests/tacacs/test_tacacs.py:137]: @rewati-arista  Should we skip this test case too?

File: sample_network_tests/tacacs/test_tacacs.py [hunk 2]
```diff
@@ -214,6 +215,7 @@ def test_if_tacacs_is_receiving_messages_on_(self, dut, tests_definitions):
...
```
Comment (kgrozis-arista) [inline sample_network_tests/tacacs/test_tacacs.py:218]: @rewati-arista Should we skip this test case too?

**Review by grybak-arista — APPROVED** [review]
Looks good, pending Kevins comments

File: sample_network_tests/tacacs/test_tacacs.py (hunk 1 above)
Comment (rewati-arista) [reply sample_network_tests/tacacs/test_tacacs.py:137]: Yep makes sense to do that, will give more visibility to the skip functionality within Vane

File: sample_network_tests/tacacs/test_tacacs.py (hunk 2 above)
Comment (rewati-arista) [reply sample_network_tests/tacacs/test_tacacs.py:216]: 0d5794430abee68c1587c8d0a28894ad2bc0c460

File: sample_network_tests/tacacs/test_tacacs.py (hunk 1 above)
Comment (rewati-arista) [reply sample_network_tests/tacacs/test_tacacs.py:134]: 0d5794430abee68c1587c8d0a28894ad2bc0c460

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**